x-pack/plugins/maps/public/embeddable/map_embeddable_factory.ts

```diff
@@ -7,23 +7,56 @@
 import _ from 'lodash';
 import { i18n } from '@kbn/i18n';
 import { IIndexPattern } from 'src/plugins/data/public';
-// @ts-ignore
-import { getMapsSavedObjectLoader } from '../angular/services/gis_map_saved_object_loader';
-import { MapEmbeddable, MapEmbeddableInput } from './map_embeddable';
-import { getIndexPatternService, getHttp, getMapsCapabilities } from '../kibana_services';
 import {
   EmbeddableFactoryDefinition,
   IContainer,
 } from '../../../../../src/plugins/embeddable/public';
+import '../index.scss';
 
-import { createMapPath, MAP_SAVED_OBJECT_TYPE, APP_ICON } from '../../common/constants';
+let whenModulesLoadedPromise = null;
+
+let getMapsSavedObjectLoader = null;
+let MapEmbeddable;
+let MapEmbeddableInput = null;
+let getIndexPatternService;
+let getHttp;
+let getMapsCapabilities = null;
+let createMapPath;
+let MAP_SAVED_OBJECT_TYPE;
+let APP_ICON = null;
+let createMapStore = null;
+let addLayerWithoutDataSync = null;
+let getQueryableUniqueIndexPatternIds = null;
+let getInitialLayers = null;
+let mergeInputWithSavedMap = null;
+
+async function loadMapDependencies() {
+  if (whenModulesLoadedPromise !== null) {
+    return whenModulesLoadedPromise;
+  }
 
-import { createMapStore } from '../reducers/store';
-import { addLayerWithoutDataSync } from '../actions/map_actions';
-import { getQueryableUniqueIndexPatternIds } from '../selectors/map_selectors';
-import { getInitialLayers } from '../angular/get_initial_layers';
-import { mergeInputWithSavedMap } from './merge_input_with_saved_map';
-import '../index.scss';
+  whenModulesLoadedPromise = new Promise(async resolve => {
+    const lazyModule = await import('./lazy');
+
+    getMapsSavedObjectLoader = lazyModule.getMapsSavedObjectLoader;
+    getQueryableUniqueIndexPatternIds = lazyModule.getQueryableUniqueIndexPatternIds;
+    MapEmbeddable = lazyModule.MapEmbeddable;
+    MapEmbeddableInput = lazyModule.MapEmbeddableInput;
+    getIndexPatternService = lazyModule.getIndexPatternService;
+    getHttp = lazyModule.getHttp;
+    getMapsCapabilities = lazyModule.getMapsCapabilities;
+    createMapPath = lazyModule.createMapPath;
+    MAP_SAVED_OBJECT_TYPE = lazyModule.MAP_SAVED_OBJECT_TYPE;
+    APP_ICON = lazyModule.APP_ICON;
+    createMapStore = lazyModule.createMapStore;
+    addLayerWithoutDataSync = lazyModule.addLayerWithoutDataSync;
+    getInitialLayers = lazyModule.getInitialLayers;
+    mergeInputWithSavedMap = lazyModule.mergeInputWithSavedMap;
```

Comment:
It would clean this up a bit written as:

```
({
    getMapsSavedObjectLoader,
    getQueryableUniqueIndexPatternIds,
    MapEmbeddable,
    MapEmbeddableInput,
    getIndexPatternService,
    getHttp,
    getMapsCapabilities,
    createMapPath,
    MAP_SAVED_OBJECT_TYPE,
    APP_ICON,
    createMapStore,
    addLayerWithoutDataSync,
    getInitialLayers,
    mergeInputWithSavedMap
} = await import('./lazy'));
```